### PR TITLE
Add an api endpoint for people to find the latest public export.

### PR DESCRIPTION
**Pull request by jfly**
This got semi-involved, so I decided to make it a separate PR, rather than adding it to #3381.

@lgarron, could you take a look?

You can see this in action over on https://staging.worldcubeassociation.org/api/v0/export/public

### CODE REVIEW

File: WcaOnRails/app/controllers/api/v0/api_controller.rb [hunk 1]
```diff
@@ -140,6 +140,22 @@ def records
     render json: json
   end
 
+  def export_public
+    sql_zips = Dir.glob(Rails.root.join("../webroot/results/misc/*.sql.zip")).sort!
```
Comment (lgarron) [inline WcaOnRails/app/controllers/api/v0/api_controller.rb]: How many of these do we have, again? It seems to return at least as fast as other HTTP requests, so I'm not super concerned.

Comment (jfly) [reply WcaOnRails/app/controllers/api/v0/api_controller.rb]: I think we have at most 2 of these, as the PHP code deletes the old export once it's done generating the next export: https://github.com/thewca/worldcubeassociation.org/blob/00b66534978cd467eff5c0ecef0e12971deb2bcf/webroot/results/admin/export_public.php#L232-L235.

**Comment by jfly** [comment]
> Should this also be added to https://staging.worldcubeassociation.org/api/v0/ ?

Oh lol, that page. I'm indifferent about updating it, as it's already quite out of date. I feel like it would be better to mention it over on https://staging.worldcubeassociation.org/results/misc/export.html as part of your work in https://github.com/thewca/worldcubeassociation.org/pull/3381.

**Comment by lgarron** [comment]
> > Should this also be added to https://staging.worldcubeassociation.org/api/v0/ ?
> 
> Oh lol, that page. I'm indifferent about updating it, as it's already quite out of date. I feel like it would be better to mention it over on https://staging.worldcubeassociation.org/results/misc/export.html as part of your work in #3381.

Alright, sounds fine to me!

**Comment by lgarron** [comment]
> > > Should this also be added to https://staging.worldcubeassociation.org/api/v0/ ?
> > 
> > 
> > Oh lol, that page. I'm indifferent about updating it, as it's already quite out of date. I feel like it would be better to mention it over on https://staging.worldcubeassociation.org/results/misc/export.html as part of your work in #3381.
> 
> Alright, sounds fine to me!

Added to https://github.com/thewca/worldcubeassociation.org/pull/3625

**Pull request merged**